django-backend/fecfiler/devops/utils/locust_data_generator.py

```diff
@@ -163,3 +166,78 @@ def generate_triple_schedule_a_transactions(self, count, reports, contacts):
         )
 
         return tier1_transactions
+
+    def generate_single_schedule_b_transactions(self, count, reports, contacts):
+        schedule_b_list = []
+        transaction_list = []
+        report_transaction_list = []
+        for _ in range(count):
+            transaction_type_identifier = "OPERATING_EXPENDITURE"
+            contact = choice(contacts)
+            report = choice(reports)
+            contribution_date = report.coverage_from_date
+            contribution_amount = randrange(25, 10000)
+            aggregation_group = "GENERAL"
+            form_type = "SB21B"
+
+            schedule_b_list.append(
+                ScheduleB(
+                    **{
+                        "expenditure_date": contribution_date,
+                        "expenditure_amount": contribution_amount,
+                        # Store report ID in purpose field
+                        "expenditure_purpose_descrip": report.id,
+                    }
+                )
+            )
+
+        schedule_b_list = ScheduleB.objects.bulk_create(schedule_b_list)
+        for schedule_b in schedule_b_list:
+            transaction_list.append(
+                Transaction(
+                    **{
+                        "transaction_type_identifier": transaction_type_identifier,
+                        "committee_account_id": self.committee.id,
+                        "contact_1_id": contact.id,
+                        "aggregation_group": aggregation_group,
+                        "form_type": form_type,
+                        "schedule_b_id": schedule_b.id,
+                    }
+                )
+            )
+
+        transaction_list = Transaction.objects.bulk_create(transaction_list)
+
+        for transaction in transaction_list:
+            report_transaction_list.append(
+                ReportTransaction(
+                    **{
+                        # Retrieve report ID from purpose field
+                        "report_id": transaction.schedule_b.expenditure_purpose_descrip,
+                        "transaction_id": transaction.id,
+                    }
+                )
+            )
+        ReportTransaction.objects.bulk_create(report_transaction_list)
+
+        return transaction_list
+
+    def generate_triple_schedule_b_transactions(self, count, reports, contacts):
+        tier1_transactions = self.generate_single_schedule_b_transactions(
+            count, reports, contacts
+        )
+        tier2_transactions = self.generate_single_schedule_b_transactions(
+            count, reports, contacts
+        )
+        tier3_transactions = self.generate_single_schedule_b_transactions(
```

Comment:
Does Schedule B have any tier 3 transactions? I know Schedule A and Schedule C do (and maybe it doesn't matter) but I wonder if we would want to remove this if not.

Reply:
From looking through our transaction models, it does not _appear_ we have any, but not 100% sure.

Reply:
thanks @lbeaufort .  On second thought it might just add to code complexity for not much gain, so totally up to you.  Also the mention of joint fundraising transfer out with reattribution is good enough reason for me to leave it alone as-is, so totally up to you.

Reply:
@dheitzer after syncing up, I [renamed triple to tiered, and removed sch B tier 3](https://github.com/fecgov/fecfile-web-api/pull/1637/commits/737b5a765f9e448ad74e20a8f71147e9c5214dcf)